bark/world/objects/agent.cpp

```diff
@@ -130,6 +130,16 @@ bool Agent::AtGoal() const {
   return goal_definition_->AtGoal(*this);
 }
 
+bool Agent::InsideRoadCorridor() const {
+  if (!road_corridor_) {
+    return false;
+  } else {
+    Polygon agent_poly = GetPolygonFromState(GetCurrentState());
```

Comment:
as a side not: I think it would be great to have such a function integrated in the agent class